src/routes/insights/Insights.tsx

```diff
@@ -1,16 +1,22 @@
-import { Hammer } from '@phosphor-icons/react';
 import classes from './Insights.module.scss';
+import { useLiveQuery } from 'dexie-react-hooks';
+import { db } from '../../data/Database';
+import DistributionChart from './DistributionChart';
+import OverviewStats from './OverviewStats';
 
 const Insights = () => {
-    return (
-        <div className={classes.insights}>
-            <h1 className={classes.header}>Insights</h1>
-            <div className={classes.content}>
-                <h2>Work in progress</h2>
-                <Hammer size={48} />
-            </div>
-        </div>
-    );
+  const activities = useLiveQuery(() => db.activities.toArray()) ?? [];
+  const categories = useLiveQuery(() => db.categories.toArray()) ?? [];
```

Comment:
Make a context of these and use the provider as the root of this component instead of prop drilling